khmer/kfile.py

```diff
@@ -167,3 +168,43 @@ def check_valid_file_exists(in_files):
         else:
             print('WARNING: Input file %s not found' %
                   in_file, file=sys.stderr)
+
+
+def is_block(fthing):
+    """Take in a file object and checks to see if it's a block or fifo."""
+    if fthing is sys.stdout or fthing is sys.stdin:
+        return True
+    else:
+        mode = os.stat(fthing.name).st_mode
+        return S_ISBLK(mode) or S_ISCHR(mode)
+
+
+def add_output_compression_type(parser):
+    """Add compression arguments to a parset object."""
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--gzip', default=False, action='store_true',
+                       help='Compress output using gzip')
+    group.add_argument('--bzip', default=False, action='store_true',
+                       help='Compress output using bzip2')
+
+
+def get_file_writer(file_handle, do_gzip, do_bzip):
+    """Generate and return a file object with specified compression."""
+    ofile = None
+
+    if do_gzip and do_bzip:
+        raise Exception("Cannot specify both bzip and gzip compression!")
+
+    if is_block(file_handle):
+        pass
+    else:
+        assert type(file_handle) == file, type(file_handle)  # Sanity check
```

Comment:
On Tue, Jul 28, 2015 at 10:25:29AM -0700, Jake Fenton wrote:
> > +    if is_block(file_handle):
> > +        pass
> > +    else:
> > +        assert type(file_handle) == file, type(file_handle)  # Sanity check

also for future I would say you should write this as

```
if not is_block(file_handle): ...
```